### DISCUSSION TITLE
How to manually set a trace in a OTEL span context ?

### DISCUSSION THREAD

**Discussion by rober15** [open-telemetry/community#1852] [Community Discussions]
As the trace propagation does not properly work via message brokers (pub sub) as lack of OTEL standards and lack of auto instrumentation i am looking now for a solution i can read out the trace id from e.g. message custom attributes and set that on my own on the message consumer inside the context which is used for logging.

Like:
- Receive message from broker
- Read out the traceparent from custom attribute
- Extract from traceparent the trace_id
- Set on trace context on receiver the context to use the trace_id received via message

Many thanks for your help!

**Comment by trask** [comment]
hi @rober15!

check out https://github.com/open-telemetry/semantic-conventions/blob/main/docs/messaging/README.md for semantic conventions related to messaging

for your other question, I'd ask in the language-specific community for the best answer
